common/test/acceptance/pages/lms/instructor_dashboard.py

```diff
@@ -685,12 +685,47 @@ class DataDownloadPage(PageObject):
     def is_browser_on_page(self):
         return self.q(css='a[data-section=data_download].active-section').present
 
+    @property
+    def generate_student_profile_report_button(self):
+        """
+        Returns the "Download profile information as a CSV" button.
+        """
+        return self.q(css='input[name=list-profiles-csv]')
+
+    @property
+    def generate_grade_report_button(self):
+        """
+        Returns the "Generate Grade Report" button.
+        """
+        return self.q(css='input[name=calculate-grades-csv]')
+
+    @property
+    def generate_weighted_problem_grade_report_button(self):
+        """
+        Returns the "Generate Weighted Problem Grade Report" button.
+        """
+        return self.q(css='input[name=problem-grade-report]')
+
+    @property
+    def report_download_links(self):
+        """
+        Returns the download links for the current page.
+        """
+        return self.q(css="#report-downloads-table .file-download-link>a")
+
+    def wait_for_available_report(self):
+        """
+        Waits for a downloadable report to be available.
+        """
+        EmptyPromise(
+            lambda: len(self.report_download_links) >= 1, 'Waiting for downloadable report'
```

Comment:
As we discussed, there are problems in both directions:
1. If there is already a report, it will return immediately with the wrong one
2. If two reports are generated, it could return both of them

In both cases, I think a clear diff of the reports in the test will make it clear what has happened. Verifying that the report received is the report expected is key to this, so I'll update the test to do this.